### PR TITLE
:art: Question buttons block redesign

### LINKED ISSUE #622: :art: Question buttons block: block redesign
## User stories

As a user, I want the "Question Block" component in the app to be redesigned to enhance its usability and visual appeal.

---

Quick links: [Summary](#summary) | [Description](#description) | [Template](#template) | [Example](#example) | [Resources](#resources)

---

### Technical Description
The current design is: 
image

It should be changed to:
image

The files can be found at `libs/features/convs-mgr/stories/blocks/library/question-message-block/`. 

The designs are [here](https://www.figma.com/file/ZGHzSstXOYPMFU1EiqjEHD/Untitled?type=design&node-id=1%3A4&mode=design&t=Hf7DxaUUMQ3BWy57-1)

### Acceptance Criteria

1. The block should look exactly like the designs.
2. Use of `rem`, `em` as opposed to `px`.
3. All changes should be isolated to the following  `libs/features/convs-mgr/stories/blocks/library/question-message-block/`.

### PR DESCRIPTION
**Pull request by darlene9497**
# Description

This PR addresses the redesign of the "Question buttons block", enhancing the usability and visual appeal of the component.

Fixes #622 

# Screenshot
Screenshot from 2023-10-12 10-26-25
Screenshot from 2023-10-11 21-15-38

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: libs/features/convs-mgr/stories/blocks/library/block-options/src/lib/components/option-input-field/option-input-field.component.html [hunk 1]
```diff
@@ -3,8 +3,8 @@
     <div formArrayName="options" fxFlex>
       <div [formGroupName]="formGroupNameInput" class="option" fxFlex>
         <input [id]="inputUniqueId" [class]="optionClass" [readOnly]="isReadOnly"
-        [ngClass]="isReadOnly? 'readonly' : ''" formControlName="message"
-          type="text" placeholder="Click here to edit" [maxlength]="charMaxlength" fxFlexFill>
+        [ngClass]="isReadOnly ? 'readonly' : ''" formControlName="message"
+        type="text" [placeholder]="'Option ' + optionIndex" [maxlength]="charMaxlength" fxFlexFill>
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/block-options/src/lib/components/option-input-field/option-input-field.component.html:7]: Let's leave the placeholder as click here to edit.

File: libs/features/convs-mgr/stories/blocks/library/block-options/src/lib/components/option-input-field/option-input-field.component.ts [hunk 2]
```diff
@@ -22,6 +22,8 @@ export class OptionInputFieldComponent implements OnInit, AfterViewInit {
 
   inputUniqueId: string;
   optionValue: string = "";
+  /**Adding this index to keep track ot the Option's index */
+  optionIndex: number;
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/block-options/src/lib/components/option-input-field/option-input-field.component.ts:27]: Let's lose the option index, don't mutate the message value when adding new option

File: libs/features/convs-mgr/stories/blocks/library/block-options/src/lib/components/option-input-field/option-input-field.component.ts [hunk 3]
```diff
@@ -30,6 +32,8 @@ export class OptionInputFieldComponent implements OnInit, AfterViewInit {
     if (!this.isNotEndpoint) {
       this.inputUniqueId = `i-${this.formGroupNameInput}-${this.blockFormGroup.value.id}`;
     }
+    /**Determine the option index and add 1 to make it 1-based instead of 0-based*/
+    this.optionIndex = +this.formGroupNameInput + 1;
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/block-options/src/lib/components/option-input-field/option-input-field.component.ts:36]: Let's lose this.

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.html [hunk 4]
```diff
@@ -7,12 +7,23 @@ <h4 class="block-title"> {{ blockTitle | transloco }}</h4>
...
+                <mat-menu #menu="matMenu">
+                    <button mat-menu-item (click)="editBlock()">
+                        <i class="far fa-edit"></i> Edit
+                    </button>
+                    <button mat-menu-item (click)="copyblock(block)">
+                        <i class="far fa-clone"></i> Copy
+                    </button>
+                    <button mat-menu-item (click)="deleteBlock()">
+                        <i class="far fa-trash-alt"></i> Delete
+                    </button>
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.html:19]: Add this to translations (Edit, Copy and Delete) and then wrap them in span tags then add a spacing of 5px

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/questions-block.jsplumb.ts [hunk 5]
```diff
@@ -21,11 +21,9 @@ export function _QuestionsBlockDecoratePlumb(block: QuestionMessageBlock,comp: C
     // Whether the anchor is target (Other Block -> This Block)
     target: true,
     cssClass:"block_endpoint",
-
-    endpoint: "Dot",
-    anchor: "Left",
+    endpoint: 'Dot',
+    anchor: [0, 0.1, 0, 1],
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/questions-block.jsplumb.ts:25]: Good

File: libs/features/convs-mgr/stories/blocks/library/question-message-block/src/lib/components/questions-block/questions-block.component.ts [hunk 6]
```diff
@@ -53,10 +53,22 @@ export class QuestionsBlockComponent implements OnInit, AfterViewInit {
     })
   }
 
+  /**ensure the option input field automatically updates */
+  setOptionMessages() {
+    this.options.controls.forEach((option, index) => {
+      option.get('message')?.setValue(`Option ${index + 1}`);
+    });
+  }
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/question-message-block/src/lib/components/questions-block/questions-block.component.ts:61]: Let's lose this method and anywhere it's called (It mutates the value of our form which is not ideal especially if the user has entered values and wants to save his/her content),

 just set a placeholder of `click here to edit` or `option`

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Very good first attempt, just some cosmetic changes left and then we can merge this.

**Comment by darlene9497** [comment]
@LemmyMwaura I have made the requested changes, kindly review and refer to the screenshots

**Comment by darlene9497** [comment]
@LemmyMwaura I have made the changes

**Review by LemmyMwaura — APPROVED** [review]

**Pull request merged**